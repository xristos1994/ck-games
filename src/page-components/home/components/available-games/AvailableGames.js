import React from "react";
import { Link } from "gatsby";
import { classnames } from "@utils/component-utils";
import * as styles from "./styles.module.css";
import { Button } from "@components";

const availableGames = [{ name: "tik tak boom", url: "/tik-tak-boom" }];

const AvailableGames = () => {
  const games = availableGames.map(({ url, name }) => (
    <Link to={url} key={name}>
      <Button
        className={classnames(styles.gameButton, "primary", "extraLargeText")}
      >
        {name}
      </Button>
    </Link>
  ));

  return <div className={styles.availableGamesContainer}>{games}</div>;
};

export { AvailableGames };
