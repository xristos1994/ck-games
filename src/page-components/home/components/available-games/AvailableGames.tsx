import React, { FC, ReactElement } from "react";
import { Link } from "gatsby";
import { classnames } from "./../../../../utils/component-utils"; // Alias "@utils/component-utils";
import { Button } from "./../../../../components"; // Alias "@components";
import { availableGames } from "./config";
const styles = require("./styles.module.css");

const AvailableGames: FC = (): ReactElement => {
  const games = availableGames.map(({ url, name }) => (
    <Link to={url} key={name}>
      <Button
        className={classnames(styles.gameButton, "primary", "extraLargeText")}
      >
        {name}
      </Button>
    </Link>
  ));

  return <div className={styles.availableGamesContainer}>{games}</div>;
};

export { AvailableGames };
