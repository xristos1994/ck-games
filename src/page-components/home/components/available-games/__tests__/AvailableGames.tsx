import React from 'react';
import renderer from 'react-test-renderer';
import { _AvailableGames } from './../AvailableGames';

jest.mock('./../config', () => ({
  __esModule: true,
  availableGames: [
    { name: 'Tik-Tak-Boom', url: '/tik-tak-boom' },
    { name: 'Pantomime', url: '/pantomime' }
  ]
}));

describe('page-components/home/AvailableGames', () => {
  it('renders correctly', () => {
    global.window = {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      location: {
        pathname: ''
      }
    };

    const t = (name: string) => name;

    const tree = renderer.create(<_AvailableGames t={t} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
