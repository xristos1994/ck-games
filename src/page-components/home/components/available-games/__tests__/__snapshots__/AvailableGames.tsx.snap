// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-components/home/AvailableGames renders correctly 1`] = `
<div
  className="availableGamesContainer"
>
  <a
    href="/tik-tak-boom"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <button
      className="main-button-hover-effect gameButton"
    >
      Tik-Tak-Boom
    </button>
  </a>
  <a
    href="/pantomime"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <button
      className="main-button-hover-effect gameButton"
    >
      Pantomime
    </button>
  </a>
</div>
`;
